gr-qtgui/lib/TimeRasterDisplayPlot.cc

```diff
@@ -38,15 +38,38 @@ namespace pt = boost::posix_time;
  **********************************************************************/
 class QwtXScaleDraw : public QwtScaleDraw, public TimeScaleData
 {
+protected:
+    double d_start_value;
+    double d_end_value;
+    double d_delta_value;
+    int d_max_scale;
+
 public:
-    QwtXScaleDraw() : QwtScaleDraw(), TimeScaleData() {}
+    QwtXScaleDraw(double start_value = 0.0, double end_value = 0.0, int max_scale = 1024)
+        : QwtScaleDraw(),
+          TimeScaleData(),
+          d_start_value(start_value),
+          d_end_value(end_value),
+          d_max_scale(max_scale)
+    {
+        d_delta_value = d_end_value - d_start_value;
+    }
+
 
     virtual ~QwtXScaleDraw() {}
 
     virtual QwtText label(double value) const
     {
-        double secs = double(value * getSecondsPerLine());
-        return QwtText(QString::number(secs, 'f', 2));
+        if (d_start_value == d_end_value) {
+            // no scale was provided.  Default to seconds.
+            double secs = double(value * getSecondsPerLine());
+            return QwtText(QString::number(secs, 'f', 2));
+        } else {
+            // User-defined scale provided.
+            double x_label =
+                d_start_value + (double)value / (double)d_max_scale * d_delta_value;
+            return QwtText(QString("").sprintf("%.2e", x_label));
```

Comment:
What happens with "%.2e" when X is large numbers, e.g., a frequency?

Reply:
I mirrored one of the other blocks to keep it similar, but I'm with you.  I'm not a fan of the fixed %.2e.  Frequencies come out as e+9 kind of numbers.  I'm going to play with some options tonight and see if I can put together a better solution.